### ISSUE TITLE
Sharing address book to groups is not working 

### ISSUE THREAD

**Issue by elhananjair** [nextcloud/contacts#2411]
### Describe the bug
......
I am having a problem with sharing address book to groups, after I share the address book and reload the whole page, the group I shared won't listed and the group doesn't get the share too.
......

**To Reproduce**
1. Go to contacts
2. Click on Settings 
3. On address book click on share option and search for a group you want to share
4. After the share reload the page, the shared group will disappear and the group or user won't get the shared address book too 

**Expected behavior**
The group must have been there for the shared group or user

**Actual behavior**
It wont be shared 

**Screenshots**

https://user-images.githubusercontent.com/69236638/131080910-947b5e1b-d8b3-4b4f-a029-eaeea8c1c173.mp4

### Server configuration

**Operating system**: Windows, Linux, MacOS...
Linux
**Web server:** Apache, Nginx...
Apache
**Database:** MariaDB, MySQL, PostgreSQL... 
PostgreSQL
**PHP version:** 7.0, 7.1, 7.2...
7.4.22
**Nextcloud version:** (see Nextcloud admin page)
21.0.4
**Contacts version:** (see Nextcloud apps page)
4.0.2
**Updated from an older Nextcloud or fresh install:**
older Nextcloud
**Signing status:**
```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

**List of activated apps:**
```
Enabled:
  - accessibility: 1.7.0
  - activity: 2.14.3
  - admin_audit: 1.11.0
  - analytics: 3.6.0
  - announcementcenter: 5.0.0
  - appointments: 1.10.0
  - apporder: 0.13.0
  - calendar: 2.3.2
  - camerarawpreviews: 0.7.12
  - circles: 0.21.4
  - cloud_federation_api: 1.4.0
  - cms_pico: 1.0.15
  - comments: 1.11.0
  - contacts: 4.0.2
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.17.1
  - deck: 1.4.3
  - duplicatefinder: 0.0.11
  - event_update_notification: 1.3.0
  - external: 3.8.2
  - extract: 1.3.2
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_automatedtagging: 1.11.0
  - files_markdown: 2.3.4
  - files_mindmap: 0.0.25
  - files_pdfviewer: 2.1.0
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_texteditor: 2.14.0
  - files_trackdownloads: 1.10.0
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - firstrunwizard: 2.10.0
  - flow_notifications: 1.1.0
  - forms: 2.3.0
  - geoblocker: 0.5.1
  - groupfolders: 9.0.2
  - integration_github: 1.0.0
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - maps: 0.1.9
  - music: 1.2.1
  - nextcloud_announcements: 1.10.0
  - notifications: 2.9.0
  - notify_push: 0.2.2
  - oauth2: 1.9.0
  - password_policy: 1.11.0
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - quicknotes: 0.7.2
  - quota_warning: 1.11.0
  - rainloop: 7.1.2
  - recommendations: 1.0.0
  - registration: 1.3.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - spreed: 11.3.1
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - tasks: 0.14.1
  - text: 3.2.0
  - theming: 1.12.0
  - theming_customcss: 1.9.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflow_ocr: 1.21.1
  - workflow_pdf_converter: 1.6.0
  - workflow_script: 1.6.0
  - workflowengine: 2.3.0
Disabled:
  - encryption
  - files_external
  - user_ldap
```

**Nextcloud configuration:**
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextacso.com",
            "www.nextacso.com",
            "https:\/\/www.nextacso.com",
            "https:\/\/nextacso.com",
            "192.168.40.2"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "21.0.4.1",
        "overwrite.cli.url": "https:\/\/nextacso.com\/",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "filelocking.enabled": true,
        "skeletondirectory": "",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "allow_local_remote_servers": true,
        "default_phone_region": "ET",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "app_install_overwrite": [
            "music",
            "pdfdraw",
            "dashboardcharts",
            "whiteboard"
        ],
        "maintenance": false,
        "loglevel": 2,
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "twofactor_enforced": "false",
        "twofactor_enforced_groups": [],
        "twofactor_enforced_excluded_groups": [],
        "mail_smtpsecure": "ssl"
    }
}
```

### Client configuration
**Browser:**
Firefox
**Operating system:**
Fedora 34
**CardDAV-clients:**

#### *Web server error log*
```
Insert your webserver log here
```

#### *data/nextcloud.log*
```
{"reqId":"YShyH3490-n7rM7-p_1XHwAAURM","level":4,"time":"2021-08-27T05:03:27+00:00","remoteAddr":"192.168.22.39","user":"Amanuel Kumera","app":"webdav","method":"REPORT","url":"/remote.php/dav/addressbooks/users/Amanuel%20Kumera/z-app-generated--contactsinteraction--recent/","message":{"Exception":"Sabre\\VObject\\ParseException","Message":"This parser only supports VCARD and VCALENDAR files","Code":0,"Trace":[{"file":"/var/www/html/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","line":90,"function":"parseDocument","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/vobject/lib/Reader.php","line":47,"function":"parse","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/CardDAV/Plugin.php","line":801,"function":"read","class":"Sabre\\VObject\\Reader","type":"::"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/CardDAV/Plugin.php","line":459,"function":"convertVCard","class":"Sabre\\CardDAV\\Plugin","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/CardDAV/Plugin.php","line":194,"function":"addressbookQueryReport","class":"Sabre\\CardDAV\\Plugin","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"report","class":"Sabre\\CardDAV\\Plugin","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":685,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpReport","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/html/nextcloud/apps/dav/lib/Server.php","line":332,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/html/nextcloud/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/var/www/html/nextcloud/remote.php","line":167,"args":["/var/www/html/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/html/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","Line":163,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0","version":"21.0.4.1","id":"61288582c9f2a"}
```

#### Browser log
<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CTRL + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
GEThttps://nextacso.com/avatar/Registration%252C%2bMonitoring%2b%2526%2bEvaluation%2bDirectorate/32
[HTTP/2 404 Not Found 1462ms]

XHRGEThttps://nextacso.com/ocs/v2.php/apps/user_status/api/v1/statuses/Registration%252C%2bMonitoring%2b%2526%2bEvaluation%2bDirectorate
[HTTP/2 404 Not Found 952ms]

XHRGEThttps://nextacso.com/ocs/v2.php/apps/user_status/api/v1/statuses/{userId}
[HTTP/2 404 Not Found 90ms]

XHRGEThttps://nextacso.com/ocs/v2.php/apps/user_status/api/v1/statuses/{userId}
[HTTP/2 404 Not Found 104ms]

XHRGEThttps://nextacso.com/ocs/v2.php/apps/user_status/api/v1/statuses/Senait%20Teame
[HTTP/2 404 Not Found 146ms]

XHRGEThttps://nextacso.com/ocs/v2.php/apps/user_status/api/v1/statuses/IT%2bTeam
[HTTP/2 404 Not Found 158ms]

GEThttps://nextacso.com/avatar/Senait%20Teame/32
[HTTP/2 404 Not Found 183ms]

GEThttps://nextacso.com/avatar/IT%2bTeam/32
[HTTP/2 404 Not Found 123ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/CA919040-EC1B-4F74-8638-AFC7DE330C3C.vcf?photo
[HTTP/2 404 Not Found 175ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/E657CE46-8068-4315-A7F9-4735086C05CC.vcf?photo
[HTTP/2 404 Not Found 156ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/CA2980FA-3777-4913-9100-E9A69176FD56.vcf?photo
[HTTP/2 404 Not Found 174ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/BD71988F-BFE8-407A-977A-BDF92CAF47E2.vcf?photo
[HTTP/2 404 Not Found 258ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/D5EB5FB7-A8CB-495D-A0FD-688E720A2B72.vcf?photo
[HTTP/2 404 Not Found 234ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/D05377C9-EE28-45FE-9A84-1E007E8280DF.vcf?photo
[HTTP/2 404 Not Found 401ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/8BA97C2D-DA94-4377-8427-9EB15773E757.vcf?photo
[HTTP/2 404 Not Found 293ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/30D2D734-8446-4570-B558-3E909DB1CA74.vcf?photo
[HTTP/2 404 Not Found 256ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/A2FDAEDC-B6BF-4F8A-9D4F-9D7FDCD95147.vcf?photo
[HTTP/2 404 Not Found 411ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/B64D6221-0DE0-4046-A4B6-7DC73E476E69.vcf?photo
[HTTP/2 404 Not Found 362ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/FE0A3073-9611-43E9-9633-326F996957F6.vcf?photo
[HTTP/2 404 Not Found 406ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/3E41AE99-8A9F-4D96-A997-02CF93EA1709.vcf?photo
[HTTP/2 404 Not Found 136ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/4EBB6865-091A-41E0-9752-CA10155C2142.vcf?photo
[HTTP/2 404 Not Found 93ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/89E08346-219A-4CB4-B138-95FF2002C4E6.vcf?photo
[HTTP/2 404 Not Found 151ms]

GEThttps://nextacso.com/remote.php/dav/addressbooks/users/administrator/contacts/024B82E7-8DFF-4C63-ABB2-30715FE1FDA6.vcf?photo
[HTTP/2 404 Not Found 180ms]

Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
content-utils.js:47

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by Hendrick42** [comment]
I have the same problem here - but a little different.

If i share with Groups from LDAP which don't have a space in the name it works.
But using Groups like "IT Persons" the share will be gone after a reload like you showed in the video.


Contacts App 4.1.0, Nextcloud 23.0.2

**Comment by max65482** [comment]
I can confirm this issue with group names that contain a space. I suspect that spaces cause mismatches (represented as `%20` in the group name preview).

**Comment by Hendrick42** [comment]
> I can confirm this issue with group names that contain a space. I suspect that spaces cause mismatches (represented as `%20` in the group name preview).

I can confirm, that the space is displayed as "%20" here, too!

**Comment by max65482** [comment]
I analyzed the issue further:
The server database stores group IDs with regular spaces (`Group With Spaces`). On the client side, however, URL encodings are used (`Group%2bWith%2bSpaces`). While processing the sharing request, the server decodes the URL to `Group+With+Spaces`. This does not match the database, hence no sharing.

I feel unable to fix this as I cannot overlook all side effects. But I opened #2708 which fixes the URL encoded group names in the dropdown selection.
